src/components/common/YourCompany/OfficeAddresses.vue

```diff
@@ -510,61 +510,75 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
       if (this.getOfficeAddresses?.registeredOffice) {
         this.mailingAddress = { ...this.getOfficeAddresses.registeredOffice.mailingAddress }
         this.deliveryAddress = { ...this.getOfficeAddresses.registeredOffice.deliveryAddress }
-
-        // compare addresses to set the "inherit mailing" flag
-        // ignore Address Type since it's different
-        // ignore Address Country Description since it's not always present
-        this.inheritMailingAddress = isSame(
-          this.getOfficeAddresses.registeredOffice.mailingAddress,
-          this.getOfficeAddresses.registeredOffice.deliveryAddress,
-          ['addressType', 'addressCountryDescription', 'id']
-        )
       }
 
+      // set initial validity states
+      // these will be updated by the BaseAddress sub-components
+      this.mailingAddressValid = !isEmpty(this.mailingAddress)
+      this.deliveryAddressValid = !isEmpty(this.deliveryAddress)
+
+      // compare valid addresses to set the "inherit mailing" flag
+      // ignore Address Type since it's different
+      // ignore Address Country Description since it's not always present
+      this.inheritMailingAddress = (
+        this.mailingAddressValid &&
+        this.deliveryAddressValid &&
+        isSame(this.mailingAddress, this.deliveryAddress, ['addressType', 'addressCountryDescription', 'id'])
+      )
```

Comment:
I added the "valid" checks so that the "same as" checkbox isn't checked when the addresses objects are empty.